src/Microsoft.AspNetCore.Authentication/LoggingExtensions.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Logging
+{
+    internal static class LoggingExtensions
+    {
+        private static Action<ILogger, string, Exception> _authSchemeAuthenticated;
+        private static Action<ILogger, string, Exception> _authSchemeNotAuthenticated;
+        private static Action<ILogger, string, string, Exception> _authSchemeNotAuthenticatedWithFailure;
+        private static Action<ILogger, string, Exception> _authSchemeSignedIn;
+        private static Action<ILogger, string, Exception> _authSchemeSignedOut;
+        private static Action<ILogger, string, Exception> _authSchemeChallenged;
+        private static Action<ILogger, string, Exception> _authSchemeForbidden;
+        private static Action<ILogger, string, Exception> _userAuthorizationFailed;
+        private static Action<ILogger, string, Exception> _userAuthorizationSucceeded;
+        private static Action<ILogger, string, Exception> _userPrincipalMerged;
+        private static Action<ILogger, string, Exception> _remoteAuthenticationError;
+        private static Action<ILogger, Exception> _signInHandled;
+        private static Action<ILogger, Exception> _signInSkipped;
+        private static Action<ILogger, string, Exception> _correlationPropertyNotFound;
+        private static Action<ILogger, string, Exception> _correlationCookieNotFound;
+        private static Action<ILogger, string, string, Exception> _unexpectedCorrelationCookieValue;
+
+        static LoggingExtensions()
+        {
+            _userAuthorizationSucceeded = LoggerMessage.Define<string>(
+                eventId: 0,
+                logLevel: LogLevel.Information,
+                formatString: "Authorization was successful for user: {UserName}.");
+            _userAuthorizationFailed = LoggerMessage.Define<string>(
+                eventId: 1,
+                logLevel: LogLevel.Information,
+                formatString: "Authorization failed for user: {UserName}.");
+            _userPrincipalMerged = LoggerMessage.Define<string>(
+                eventId: 0,
```

Comment:
I wonder if it's easiest in a case like this to just have all the events by incrementing even though that makes them "more unique" than necessary?

@lodejard thoughts? The log events here are used in several different classes across the assembly. In theory each class can start its event id's at 0 and increase, but that's a huge hassle to track. It's a lot easier to just start at 0 "per assembly" or at least "per logger extensions class" and have everything be unique within that scope.